packages/babel-parser/src/parser/lval.js

```diff
@@ -383,7 +383,11 @@ export default class LValParser extends NodeUtils {
             checkClashes[key] = true;
           }
         }
-        if (bindingType === BIND_LEXICAL && expr.name === "let") {
+        if (
+          bindingType === BIND_LEXICAL &&
+          expr.name === "let" &&
+          !this.scope.inCatchBlock
+        ) {
```

Comment:
Does this approach work with this code, which should throw?

```js
try {} catch (err) {
  let let;
}
```

Reply:
It doesn’t, thanks for pointing it out, i'll work on that